conans/model/conan_file.py

```diff
@@ -59,7 +59,7 @@ def create_settings(conanfile, settings):
         settings.constraint(current)
         return settings
     except Exception as e:
-        raise ConanException("Error while initializing settings. %s" % str(e))
+        raise ConanInvalidConfiguration("Error while initializing settings. %s" % str(e))
```

Comment:
I think this is a bit dangerous, better raise ConanInvalidConfiguration inside ``settings.constraint()`` for controlled cases instead, and forward it from here.

Reply:
Nevermind, this is too complicated without deep changes in Settings. Lets leave it as-is, the path to go is have Settings return different exception types for different errors.